docs_src/design/adr/system-management/001-docker-image-guidelines.md

```diff
@@ -0,0 +1,110 @@
+# Docker image guidelines
+
+## Status
+
+Pending
+
+## Context
+
+When deploying the EdgeX Docker containers some security measures are recommended to ensure the integrity of the software stack.
+
+## Decision
+
+When deploying Docker images, the following flags should be set for heightened security.
+
+- To avoid escalation of privileges each docker container should use the `no-new-privileges` option in their Docker compose file (example below). More details about this flag can be found [here](https://docs.docker.com/engine/reference/run/#security-configuration). This follows Rule #4 for Docker security found [here](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html#rule-4-add-no-new-privileges-flag).
+
+```docker
+security_opt:
+      - "no-new-privileges:true"
+```
+
+- To further prevent privilege escalation attacks the user should be set for the docker container using the `--user=<userid>` or `-u=<userid>` option in their Docker compose file (example below). More details about this flag can be found [here](https://docs.docker.com/engine/reference/run/#user). This follows Rule #4 for Docker security found [here](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html#rule-2-set-a-user).
+
+```docker
+services:
+  device-virtual:
+    image: ${REPOSITORY}/docker-device-virtual-go${ARCH}:${DEVICE_VIRTUAL_VERSION}
+    user: 4000 # user option using an unprivileged user
+    ports:
+    - "127.0.0.1:49990:49990"
+    container_name: edgex-device-virtual
+    hostname: edgex-device-virtual
+    networks:
+      - edgex-network
+    env_file:
+      - common.env
+    environment:
+      SERVICE_HOST: edgex-device-virtual
+    depends_on:
+      - consul
+      - data
+      - metadata
+```
+
+> NOTE: exception
+    Sometimes containers will require root access to perform their fuctions. For example the System Management Agent requires root access to control other Docker containers. Inthis case you would need to assign the `user: root`.
```

Comment:
change `Inthis case you would need to assign the `user: root`.` to `In this case you would allow it run as default `root` user. Unless we want to be explicit on all container....

Reply:
got it